dictionary/views.py

```diff
@@ -1,3 +1,27 @@
+'''
+Views for the dictionary application.
+'''
+
 from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.contrib.postgres.search import SearchVector
+
+from . import forms
+from . import models
+
+def search(request):
+    '''
+    This is a first crack at a basic search interface.
+    '''
+    if request.method == 'POST':
+        form = forms.SearchForm(request.POST)
+        if form.is_valid():
+            search_text = form.cleaned_data['search_text']
+            search_vector = SearchVector('simple', 'traditional', 'pronunciation')
+            results = models.Entry.objects.annotate(search=search_vector).filter(search=search_text)
+            form = forms.SearchForm()
+            return render(request, 'dictionary/search.html', {'form': form, 'results': results,})
+    else:
+        form = forms.SearchForm()
```

Comment:
I don't think trigrams would work well for search, a trigram is three words occurring together, so it will narrow the result set rather than expand it (which is I assume what you're wanting to do with fuzzy searching). Bigrams and trigrams would be more useful perhaps in scoring search results (ordering them from most to least relevant). I believe there are other proximity measures that would be useful for scoring results. for instance if the user searched for "Chicago Bears" a bigram would recognize:

"The Chicago Bears continue their losing streak."

But would miss:

"The Bears are back in Chicago this week."

I've actually not used the postgres text search interface, I've just used ElasticSearch. So I'd have to dig through the documentation for edit distance. I know that in ElasticSearch you can specify an edit distance by adding a "fuzzy=x" parameter where 'x' is an integer specifying how many edits you want to allow. So for instance "fuzzy=2" will match any word an edit distance of 2 within any search word. So "wihtin" would match "within" (edit distance=1) and whiten (edit distance=2). I've found two edit distances to be a lot in english and three to be way too much for search purposes.

I don't mean to advocate for ES because it is a big pita monster but it also allows you to implement "Did you mean x?" functionality.